### PR TITLE
Added extra params (width, height, playerVars)

### PR DESCRIPTION
**Pull request by mfclarke**

### CODE REVIEW

**Comment by compedit** [comment]
I like how you've approached it, but I'm worried about a few of the new `playerVars`, (specifically `enablejsapi` and `autoplay`) affecting the existing API.

`autoplay` is going to have to be pulled out of the top level props for this to work, otherwise it will clash with `playerVars.autoplay`. Right now that's all being managed in `_loadUrl`, so that will have probably have to be rewritten.

Additionally, I'm not sure if `playerVars.autoplay` works after the first load so that will take some investigation.



As to how it's going to affect the external API, the props at the moment are:

```js
<YouTube 
  url={string}         
  id={string}          
  autoplay={bool}      
  onPlayerReady={func}
  onVideoReady={func} 
  onPlay={func}       
  onPause={func}      
  onEnd={func}        
/> 
```

Adding in your PR would make it 

```js
<YouTube 
  url={string}         
  id={string}          
  autoplay={bool}      
  height={string}
  width={string}
  playerVars={object}
  onPlayerReady={func}
  onVideoReady={func} 
  onPlay={func}       
  onPause={func}      
  onEnd={func}        
/> 
```

But if we edit `createPlayer` like I mentioned in the PR comments, we can have this:

```js
<YouTube 
  url={string}          
  opts={object}
  onPlayerReady={func}
  onVideoReady={func} 
  onPlay={func}       
  onPause={func}      
  onEnd={func}         
/> 
```

Basically, hell yes I want to support this but it's going to require some changes to & rethinking about the internal/external API. So I'm going close this PR and open up a new branch to work under, would love your help on it if you're down. Thanks!

**Comment by mfclarke** [comment]
Ok great, having it all in opts is nice (and I'm currently using the Soundcloud widget and like how it works there, so, awesome). I'll have a crack at this over the next few days :)
